store/src/java/com/zimbra/cs/datasource/imap/ImapFolderSync.java

```diff
@@ -350,6 +356,13 @@ public void syncMessages() throws ServiceException, IOException {
             }
         }
 
+        // ZBUG-2361: this is to push local draft changes into remote 
+        if (DRAFTS.equals(remoteFolder.getPath())) {
+            changes = MessageChanges.getChangesForDrafts(ds, localFolder.getFolder(), lastChangeId,
+                    localFolder.getId());
+            pushChangesForDraft(changes, remoteFolder.getPath());
```

Comment:
This will probably need to [handle updating the changeId](https://github.com/Zimbra/zm-mailbox/blob/9ece3daf78fd27bed7db95d27e12b9896855109a/store/src/java/com/zimbra/cs/datasource/imap/ImapFolderSync.java#L310) again if you use two separate calls to find changes on the draft folder since we could capture a later change from the second call, findChangesForDrafts.

Reply:
The tomstone wouldn't be required here as delete operation is taken care in previous scenarios, hence removed from here
I'll quickly do the test by updating the changeId , I assume we have to handle it
Thanks for this. I'll update test and let know